src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningBarnetilsynUtil.kt

```diff
@@ -0,0 +1,70 @@
+                                           tilOgMedDato = LocalDate.MAX,
+                                           maxbeløp = mapOf(1 to 4250, 2 to 5545, 3 to 6284)),
+                   MaxbeløpBarnetilsynSats(fraOgMedDato = LocalDate.parse("2021-01-01"),
+                                           tilOgMedDato = YearMonth.of(2021, 12).atEndOfMonth(),
+                                           maxbeløp = mapOf(1 to 4195, 2 to 5474, 3 to 6203)),
+                   MaxbeløpBarnetilsynSats(fraOgMedDato = LocalDate.parse("2020-01-01"),
+                                           tilOgMedDato = YearMonth.of(2020, 12).atEndOfMonth(),
+                                           maxbeløp = mapOf(1 to 4053, 2 to 5289, 3 to 5993))
+            )
+
+    fun lagBeløpsPeriodeBarnetilsyn(utgiftsperiode: UtgiftsMåned,
+                                    kontantstøtteBeløp: BigDecimal,
+                                    tilleggsstønadBeløp: BigDecimal,
+                                    antallBarnIPeriode: Int): BeløpsperiodeBarnetilsynDto {
+        val beløpPeriode: BigDecimal =
+                beregnPeriodeBeløp(utgiftsperiode.utgifter,
+                                   kontantstøtteBeløp,
+                                   tilleggsstønadBeløp,
+                                   antallBarnIPeriode,
+                                   utgiftsperiode.årMåned)
+
+        return BeløpsperiodeBarnetilsynDto(utgiftsperiode.årMåned.tilPeriode(),
+                                           beløpPeriode,
+                                           BeregningsgrunnlagBarnetilsynDto(
+                                                   utgiftsbeløp = utgiftsperiode.utgifter,
+                                                   kontantstøttebeløp = kontantstøtteBeløp,
+                                                   tilleggsstønadsbeløp = tilleggsstønadBeløp,
+                                                   antallBarn = antallBarnIPeriode))
+    }
+
+    fun beregnPeriodeBeløp(periodeutgift: BigDecimal,
+                           kontrantstøtteBeløp: BigDecimal,
+                           tillegsønadBeløp: BigDecimal,
+                           antallBarn: Int,
+                           årMåned: YearMonth) =
+            maxOf(ZERO, minOf(((periodeutgift - kontrantstøtteBeløp) * 0.64.toBigDecimal()) - tillegsønadBeløp,
+                              satserForBarnetilsyn.hentSatsFor(antallBarn, årMåned).toBigDecimal()))
+
+    private fun YearMonth.tilPeriode(): Periode {
+        return Periode(this.atDay(1),
+                       this.atEndOfMonth())
+    }
+}
+
+fun List<MaxbeløpBarnetilsynSats>.hentSatsFor(antallBarn: Int, årMåned: YearMonth): Int {
+    val maxbeløpBarnetilsynSats = this.filter {
+        it.fraOgMedDato <= årMåned.atDay(1) && it.tilOgMedDato >= årMåned.atDay(1)
+    }.singleOrNull() ?: error("Kunne ikke finne barnetilsyn sats for dato: $årMåned ")
```

Comment:
Her ger intellij warning, slå sammen filter + singleOrNull
```suggestion
    val maxbeløpBarnetilsynSats = this.singleOrNull {
        it.fraOgMedDato <= årMåned.atDay(1) && it.tilOgMedDato >= årMåned.atDay(1)
    } ?: error("Kunne ikke finne barnetilsyn sats for dato: $årMåned ")
```

Reply:
Hmmmm. Du har streng Intellij. Kanskje det er på tide at jeg oppdaterer. Får ikke noen warning. Men - uansett kortere og penere kode - så jeg fikser.